software/Test_Robot_Software/state_machine.py

```diff
@@ -0,0 +1,63 @@
+from robot_state import State
+from api import RobotMovement
+
+class StateMachine:
+    """
+    This class is responsible for switching the states
+    of the robot as implementing the state machine logic
+    """
+
+    def __init__(self, robot_movement):
+        self.robot_movement = robot_movement
+        self.state = State.FIND_A_BALL
```

Comment:
Make sure the state is only defined in this class. Other code should only modify it with methods and functionalities provided by the state_machine class

Reply:
updated the `main.py` and `state_machine.py` files